openassessment/data.py

```diff
@@ -993,15 +997,26 @@ def create_zip_with_attachments(cls, file, course_id, submission_files_data):
 
         with ZipFile(file, 'w') as zip_file:
             for file_data in submission_files_data:
-                file_content = (
-                    cls._download_file_by_key(file_data['key'])
-                    if file_data['type'] == cls.ATTACHMENT
-                    else file_data['content']
-                )
+                key = file_data['key']
+                file_name = file_data['file_path']
+
+                try:
+                    file_content = (
+                        cls._download_file_by_key(file_data['key'])
+                        if file_data['type'] == cls.ATTACHMENT
+                        else file_data['content']
+                    )
+                except FileMissingException:
+                    file_data['valid'] = False
+                    # added a header to csv file to indicate that the file in valid or not.
+                    # not sure if I should create a {file_name}.error.txt to indicate the file error more clearly.
+                    logging.warning('File: {file_name} with key: {key} was missing.'
```

Comment:
I'd also include the user, course, and the block location so that this error message can be more easily searched. Unless that's already included by default in the prefix? I don't think it is.

Reply:
course_id, block_id, and student_id should all be available to you in file_data